package/MDAnalysis/lib/nsgrid.pyx

```diff
@@ -0,0 +1,996 @@
+        ``[xmax, ymax, zmax] - [xmin, ymin, zmin]`` should be used
+        to construct a pseudo box. Subsequently, the origin should also be
+        shifted to ``[xmin, ymin, zmin]``. These arguments must be provided
+        to the function.
+
+        Parameters
+        ----------
+        cutoff : float
+            Desired cutoff distance
+        coords : numpy.ndarray
+            atom coordinates of shape ``(N, 3)`` for ``N`` atoms.
+            ``dtype=numpy.float32``. For Non-PBC calculations,
+            all the coords must be within the bounding box specified
+            by ``box``
+        box : numpy.ndarray
+            Box dimension of shape (6, ). The dimensions must be
+            provided in the same format as returned
+            by :attr:`MDAnalysis.coordinates.base.Timestep.dimensions`:
+            ``[lx, ly, lz, alpha, beta, gamma]``. For non-PBC
+            evaluations, provide an orthogonal bounding box
+            (dtype = numpy.float32)
+        max_gridsize : int
+            maximum number of cells in the grid. This parameter
+            can be tuned for superior performance.
+        pbc : boolean
+            Handle to switch periodic boundary conditions on/off [True]
+
+        Note
+        ----
+        * ``pbc=False`` Only works for orthogonal boxes.
+        * Care must be taken such that all particles are inside
+          the bounding box as defined by the box argument for non-PBC
+          calculations.
+        * In case of Non-PBC calculations, a bounding box must be provided
+          to encompass all the coordinates as well as the search coordinates.
+          The dimension should be similar to ``box`` argument but for
+          an orthogonal box. For instance, one valid set of argument
+          for ``box`` for the case of no PBC could be
+          ``[10, 10, 10, 90, 90, 90]``
+        * Following operations are advisable for non-PBC calculations
+
+        ..code-block:: python
+
+            lmax = all_coords.max(axis=0)
+            lmin = all_coords.min(axis=0)
+            pseudobox[:3] = 1.1*(lmax - lmin)
+            pseudobox[3:] = 90.
+            shift = all_coords.copy()
+            shift -= lmin
+            gridsearch = FastNS(max_cutoff, shift, box=pseudobox, pbc=False)
+
+        """
+
+        from MDAnalysis.lib.mdamath import triclinic_vectors
+
+
+        _check_array(coords, 'coords')
+
+        if np.allclose(box[:3], 0.0):
+            raise ValueError("Any of the box dimensions cannot be 0")
+
+        self.periodic = pbc
+        self.coords = coords.copy()
+
+        if box.shape != (3, 3):
+            box = triclinic_vectors(box)
+
+        self.box = PBCBox(box, self.periodic)
+
+        if cutoff < 0:
+            raise ValueError("Cutoff must be positive!")
+        if cutoff * cutoff > self.box.c_pbcbox.max_cutoff2:
+            raise ValueError("Cutoff greater than maximum cutoff ({:.3f}) given the PBC")
```

Comment:
Here the maximum cutoff is not displayed when raising the error

Reply:
Hi Charly. Thank you for reporting the problem.

A comment on a merged pull request is likely to be forgotten, though. Would you mind opening an issue?